### PR TITLE
Use Qt's builtin network proxy configuration

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

**Comment by pvuorela** [comment]
Doesn't build, libssunetworkproxy/ssunetworkproxy.h gets included in multiple places.

**Comment by rainemak** [comment]
Please update and rebase

**Review by rainemak — CHANGES_REQUESTED** [review]
As per https://github.com/sailfishos/ssu/pull/5#issuecomment-1016483467 requires changes.

**Comment by martyone** [comment]
Yup, not forgotten, originally suspended waiting for SDK build with the qtbase side change integrated (available now), will return to this soon.

**Comment by martyone** [comment]
> Doesn't build, libssunetworkproxy/ssunetworkproxy.h gets included in multiple places.

Somehow I didn't push the latest version before opening the PR. Rebased & pushed now. Now also properly tested and it needs one more change to qtbase https://github.com/sailfishos/qtbase/pull/7, so that must be merged first.

File: rpm/ssu.spec [hunk 1]
```diff
@@ -21,7 +21,6 @@ Requires(pre): shadow-utils
 Requires(pre): /usr/bin/groupadd-user
 Requires(postun): shadow-utils
 Requires: ssu-vendor-data
-Requires: ssu-network-proxy
```
Comment (pvuorela) [inline rpm/ssu.spec:24]: Some Obsoletes/Provides: here to get the package nicely uninstalled?

Comment (martyone) [reply rpm/ssu.spec:23]: Fixed. Not sure whether there are alternatives to the `network-proxy-plugin` package, providing `ssu-network-proxy`.

File: rpm/ssu.spec [hunk 2]
```diff
@@ -20,7 +20,7 @@ Requires(pre): sailfish-setup
 Requires(pre): shadow-utils
 Requires(pre): /usr/bin/groupadd-user
 Requires(postun): shadow-utils
-Requires: ssu-network-proxy
+Obsoletes: network-proxy-plugin
```
Comment (rainemak) [inline rpm/ssu.spec:23]: warning: line 23: It's not recommended to have unversioned Obsoletes: Obsoletes: network-proxy-plugin

=> Obsoletes: ssu-network-proxy <= 1.0.22

Comment (martyone) [reply rpm/ssu.spec]: I hoped we would drop the warning at some point instead :) fixed.

Comment (rainemak) [reply rpm/ssu.spec]: True that but this looks good.

**Review by rainemak — APPROVED** [review]
Thanks @martyone !

**Pull request merged**